### ISSUE TITLE
MQTT options are not set for MQTT client

### ISSUE THREAD

**Issue by AnatolyStrakhov** [hannseman/homebridge-mi-hygrothermograph#76]
The quote from the main page:

> Everything set in mqtt will be passed to the options argument on Client.

It is not true according to this piece of code.

```
  publishValueToMQTT(topic, value) {
    this.mqttClient.publish(topic, String(value), {
      qos: this.config.mqtt.qos || 0
    });
  }

```
Only qos is set.
My config has
```

 "accessories": [
     {
       "mqtt": {
         "url": "mqtt://mybroker",
         "temperatureTopic": "hello/world",
         "retain": true
       },
       "accessory": "Hygrotermograph",
       "name": "Temperature & Humidity",
       "address": "MAC address",
       "timeout": 15,
       "temperatureName": "Temperature",
       "humidityName": "Humidity",
       "fakeGatoEnabled": true
     }
   ]

```
And there is no retained messages in my broker
Only with

```
    this.mqttClient.publish(topic, String(value), {
       qos: this.config.mqtt.qos || 0,
       retain: this.config.mqtt.retain || false
     });

```
I have retained changes. 

### Expected Behavior
MQTT options from config are set.

### Current Behavior
Only qos option is set.

### Steps to Reproduce (for bugs)
1. Configure a sensor to send retained MQTT messages to a broker.
2. Configure the broker to get retained MQTT messages.
3. Send a message to the broker.
4. Check that there is no retained messages in broker's queue.

### Versions

- **homebridge**: `homebridge --version` 0.4.50
- **homebridge-mi-hygrothermograph**: `npm list --depth 1 --global homebridge-mi-hygrothermograph` 2.3.0
- **node**:       `node --version` 9.11.2
- **OS**:          Raspbian buster 2019-09-26

### Configuration

```
 "accessories": [
     {
       "mqtt": {
         "url": "mqtt://mybroker",
         "temperatureTopic": "hello/world",
         "retain": true
       },
       "accessory": "Hygrotermograph",
       "name": "Temperature & Humidity",
       "address": "MAC address",
       "timeout": 15,
       "temperatureName": "Temperature",
       "humidityName": "Humidity",
       "fakeGatoEnabled": true
     }
   ]

```

### Homebridge Log / Command Output

```

```

**Comment by hannseman** [comment]
Thanks for the detailed issue, appreciate it!

Yes you're right, that statement only holds true for arguments relevant to `client.connect`. See: https://github.com/hannseman/homebridge-mi-hygrothermograph/blob/master/lib/accessory.js#L261-L277

Since the options for `client.publish` is not as complex as those to `client.connect` maybe we can just do the same as for `qos` (same as in your fixed example).

Would you like to open up a PR adding  `retain: this.config.mqtt.retain || false` to the publish-call?

**Comment by AnatolyStrakhov** [comment]
Hi @hannseman ,
I am not the owner of the product, so it is up to you how to fix the issue. My fix is enough for me.

**Comment by hannseman** [comment]
This is an open source project, I invited you to contribute so your fix can reach everyone using this project.

**Comment by hannseman** [comment]
Fixed in v2.3.1.
